File: 프로그래머스/프로그래머스_Lv1/[PCCP 기출문제 ] 1번 /solution.js
function solution(bandage, health, attacks) {
  let leftHealth = health;
  attacks.unshift([0, 0]);
  for (let index = 1; index < attacks.length; index++) {
    let timeGap = attacks[index][0] - attacks[index - 1][0] - 1;
    let newHealth = Math.min(
      leftHealth +
        timeGap * bandage[1] +
        Math.floor(timeGap >= bandage[0] ? 1 : 0) * bandage[2],
      health
    );
    newHealth -= attacks[index][1];
    if (leftHealth <= 0) {
      return -1;
    }
    leftHealth = newHealth;
  }

  return leftHealth <= 0 ? -1 : leftHealth;
}

File: 프로그래머스/프로그래머스_Lv1/[PCCP 기출문제 ] 1번 /solution.java
class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {
    
    int cast_time = bandage[0]; 
    int healing_per_second = bandage[1]; 
    int additional_healing = bandage[2]; 
    
    int max_health = health; 
    
    int pre_attack_time = attacks[0][0];
    
    for(int i=0; i<attacks.length; i++){
        int attack_time = attacks[i][0];
        int damage = attacks[i][1];
        int time_between = attack_time - pre_attack_time - 1;
    if (time_between > 0){
    
        health += (time_between * healing_per_second);
        health += ((time_between / cast_time) * additional_healing);
    
            if (health > max_health){
                health = max_health;
            }
        }
        health -= damage;
        if (health <= 0) return -1;
        pre_attack_time = attack_time;
    }
    return health;
    }
}